plugins/Goals/API.php

```diff
@@ -331,11 +332,29 @@ protected function getItems($recordName, $idSite, $period, $date, $abandonedCart
         $archive   = Archive::build($idSite, $period, $date, $segment);
         $dataTable = $archive->getDataTable($recordNameFinal);
 
-        $this->enrichItemsTableWithViewMetrics($dataTable, $recordName, $idSite, $period, $date, $segment);
+        // Before Matomo 4.0.0 ecommerce views were tracked in custom variables
+        // So if Matomo was installed before try to fetch the views from custom variables and enrich the report
+        if (version_compare(DbHelper::getInstallVersion(),'4.0.0-b2', '<')) {
```

Comment:
btw getInstallVersion might not be set AFAIK as it was only added in 2018

Reply:
That should be fine I guess. `version_compare(null, '4.0.0-b2', '<')` returns `true`